locator/src/backup_routes.rs

```diff
@@ -116,22 +123,92 @@ impl BackupRouteProvider for FilesystemRouteProvider {
     }
 }
 
-pub struct GcsRouteProvider {}
+pub struct GcsRouteProvider {
+    bucket_name: String,
+    codec: Codec,
+    object_key: String,
+    client: google_cloud_storage::client::Storage,
+    // If the last cursor does not change, skip upload
+    last_cursor: Mutex<Option<String>>,
+}
 
 impl GcsRouteProvider {
-    pub fn new(_bucket: &str) -> Self {
-        GcsRouteProvider {}
+    pub async fn new(bucket: String) -> Result<Self, BackupError> {
+        let client = google_cloud_storage::client::Storage::builder()
```

Comment:
How does auth work? Does it rely on setting up of some environment variable? If so, we should document it as a requirement. Also, for environment variables I like having .env file in the repo so its clear what dependencies exist for a project on environment variable.

Reply:
This is something I'll probably have to test + probably develop further in the sandbox, but my understanding is that the application default credentials should get picked up automatically.

Reply:
Ok. Either ways lets document how this should work locally as well as in production. Imagine coming back to this code 6 months from now and not remembering that you need to setup something locally for it to work.